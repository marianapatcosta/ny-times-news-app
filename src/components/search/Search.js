import { defineComponent } from "vue";

export default defineComponent({
  name: "Search",
  emits: ["search"],
  data() {
    return {
      keyword: "",
    };
  },
  methods: {
    handleOnEnter(event) {
      const target = event.target;
      this.$emit("search", this.keyword);
      target.blur();
    },
  },
});
