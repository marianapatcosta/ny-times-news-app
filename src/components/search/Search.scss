.search {
  height: 2rem;
  border: 0.15rem solid $color-secondary;
  padding: 0.3rem 0.5rem;

  &:focus {
    border: 0.15rem solid $color-transparent;
  }

  &::placeholder {
    opacity: 0.8;
  }
}
