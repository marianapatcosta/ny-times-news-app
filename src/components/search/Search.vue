<template>
  <input
    class="search"
    type="search"
    v-model="keyword"
    placeholder="Search by..."
    aria-label="Search New York time News"
    @search="handleOnEnter"
    @keydown.enter="handleOnEnter"
  />
</template>

<script src="./Search.js"></script>

<style scoped lang="scss" src="./Search.scss"></style>
