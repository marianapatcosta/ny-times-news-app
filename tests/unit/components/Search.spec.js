import { shallowMount } from "@vue/test-utils";
import Search from "@/components/search/Search.vue";

describe("Search.vue", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(Search);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it("should render without errors and match snapshot renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should handle search event", async () => {
    const search = wrapper.find('input[type="search"]');
    await search.setValue("search keyword");
    await wrapper.trigger("keydown.enter");
    expect(search.element.value).toMatch("search keyword");
    expect(wrapper.emitted()).toHaveProperty("search");
    expect(wrapper.emitted().search[0]).toEqual(["search keyword"]);
  });
});
