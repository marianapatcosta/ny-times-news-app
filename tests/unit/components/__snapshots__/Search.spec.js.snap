// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search.vue should render without errors and match snapshot renders correctly 1`] = `
<input
  aria-label="Search New York time News"
  class="search"
  placeholder="Search by..."
  type="search"
/>
`;
